### DISCUSSION TITLE
如何在xmake.lua中获取到conan依赖的路径

### DISCUSSION THREAD

**Discussion by FlyinCow** [xmake-io/xmake#5976] [Q&A]
我在xmake.lua中声明了一个依赖，它安装之后的布局是这样的：
```
.
|-- bin
|   `-- Release // 虽然设置了add_rules("mode.release", "mode.debug", 但是只有Release不知道为什么)
|       |-- A.dll
|       `-- B.dll
|-- include // 一堆头文件
`-- lib // 定位lib
    `-- Release
        |-- A.lib
        `-- B.lib
```
如果我只使用add_packages是无法自动链接的，我可以手动add_links("Release/A")或者add_packages("lib", {links = "Release/A"})。有什么办法可以在这种布局下自动链接吗，或者让我根据包名获取conan依赖的位置然后手动做字符串操作？MSVC默认编译出来的布局应该就是这样的。

**Comment by FlyinCow** [comment]
以及这种情况是不是还需要add_rpathdirs？不管是add_rpathdirs还是拷贝dll，都需要获取到conan依赖的路径，似乎xmake只在conan依赖的bin目录里面找平铺的dll

**Comment by waruqi** [comment]
不懂你要干嘛，按正常包集成方式，集成下 conan 包不就好了么。。add_requires+add_packages

**Reply by FlyinCow** [reply]
这conan包里面不是bin下面就是dll的，还有一层Release/Debug文件夹，直接add_requires+add_packages是不会自动链接的，也不会设置正确的rpath。

**Reply by waruqi** [reply]
给个完整复现的 project zip，正常 `add_requires("conan::zlib")` 集成的包，安装路径都是带 hash，并且通过 conan 提供的 manifest 信息集成的，我不知道你怎么搞得

**Reply by FlyinCow** [reply]
可能给不了，是公司内部的conan package，我研究一下conan manifest看看是不是他打包的问题先。
